### PR TITLE
Add K8s Operator docs

### PR DESCRIPTION
**Pull request by peterdeme**
# Description of the change

K8s operator docs.

## Checklist

Please make sure that the proposed change checks all the boxes below before requesting a review:

- [ ] I have reviewed the [guidelines for contributing](https://github.com/spacelift-io/user-documentation/blob/main/CONTRIBUTING.md) to this repository.
- [ ] The preview looks fine.
- [ ] The tests pass.
- [ ] The commit history is clean and meaningful.
- [ ] The pull request is opened against the `main` branch.
- [ ] The pull request is no longer marked as a draft.
- [ ] You agree to license your contribution under the [MIT license](https://github.com/spacelift-io/user-documentation/blob/main/LICENSE) to Spacelift (not required for Spacelift employees).
- You have updated the navigation files correctly:
    - [ ] No new pages have been added, or;
    - [ ] Only _nav.yaml_ has been updated because the changes only apply to SaaS, or;
    - [ ] Only _nav.self-hosted.yaml_ has been updated because the changes only apply to Self-Hosted, or;
    - [ ] Both _nav.yaml_ and _nav.self-hosted.yaml_ have been updated.

If the proposed change is ready to be merged, please request a review from `@spacelift-io/solutions-engineering`. Someone will review and merge the pull request.

_Spacelift employees should request reviews from the relevant engineers and are allowed to merge pull requests after they got at least one approval._

Thank you for your contribution! 🙇

### CODE REVIEW

File: docs/integrations/kubernetes/operator.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+```
+
+!!! hint
+    You can also download the manifest file, inspect it first, and then apply it using `kubectl apply -f <path-to-manifest-file>`.
+
+This command will install the [Custom Resource Definition](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/){: rel="nofollow"} (CRD) and the operator itself.
+
+### Create a secret with your Spacelift API key
+
+You need to create an [Opaque secret](https://kubernetes.io/docs/concepts/configuration/secret/#opaque-secrets){: rel="nofollow"} with your [Spacelift API key](../api.md#spacelift-api-key--token) so the operator can authenticate with the Spacelift API. The secret's name has to be `spacelift-credentials` and it requires 3 keys:
```
Comment (eliecharra) [inline docs/integrations/kubernetes/operator.md:27]:
```suggestion
You need to create an [Opaque secret](https://kubernetes.io/docs/concepts/configuration/secret/#opaque-secrets){: rel="nofollow"} with your [Spacelift API key](../api.md#spacelift-api-key-token) so the operator can authenticate with the Spacelift API. The secret's name has to be `spacelift-credentials` and it requires 3 keys:
```

File: docs/integrations/kubernetes/operator.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+### Create a test Space
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: app.spacelift.io/v1beta1
+kind: Space
+metadata:
+  name: space-test
+spec:
+  name: Test space created from Operator
```
Comment (eliecharra) [inline docs/integrations/kubernetes/operator.md:50]: `spec.name` is now optional, maybe we should keep this example as minimal as possible to limit confusion on the name field.

```suggestion
```

File: docs/integrations/kubernetes/operator.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+⚠️ **Note**: The operator currently does not delete the Space when the CR is deleted. You'll need to manually delete the Space in Spacelift.
+
+## Debugging
+
+The operator is built with [kubebuilder](https://book.kubebuilder.io/){: rel="nofollow"} and uses [zap](https://github.com/uber-go/zap){: rel="nofollow"} for logging. The default loglevel is `info`. You can change this by editing the manifest file (look for the `public.ecr.aws/spacelift-io/spacelift-operator` container) and adding an argument to the argument list:
+
+```yaml
+args:
+  - "<existing-arguments>"
+  - "--zap-log-level=debug"
```
Comment (eliecharra) [inline docs/integrations/kubernetes/operator.md:70]: Did you tried that? I was thinking that the zap log level should be a number 🤔

**Review by eliecharra — COMMENTED** [review]
We should also add a section explaining how to find the reference for all the available fields.
I'm wondering if we should reference the go structs or the CRD YAML in `config/crd/bases` 🤔 I'm not sure what would be the easier to read from customer perspective.

Otherwise, sounds good 👍🏻

File: docs/integrations/kubernetes/operator.md (hunk 3 above)
Comment (peterdeme) [reply docs/integrations/kubernetes/operator.md:70]: I did yeah. K8s has a parsing logic for strings. :) I found the numbers confusing so I used a string here.

**Comment by peterdeme** [comment]
@eliecharra 

> I'm wondering if we should reference the go structs or the CRD YAML

image

**Comment by eliecharra** [comment]
> @eliecharra
> 
> > I'm wondering if we should reference the go structs or the CRD YAML
> 
> image

😂 Yeah sure good idea 👍🏻

File: docs/integrations/kubernetes/operator.md (hunk 3 above)
Comment (eliecharra) [reply docs/integrations/kubernetes/operator.md:70]: Good to know 😃

File: docs/integrations/kubernetes/operator.md (hunk 1 above)
Comment (peterdeme) [reply docs/integrations/kubernetes/operator.md:27]: nope, this is correct actually. :)

**Comment by peterdeme** [comment]
@eliecharra added references to the CRD, a warning section to the top, and a small troubleshooting section to the bottom

**Review by eliecharra — APPROVED** [review]

**Pull request merged**